### DISCUSSION TITLE
Full sync messages between production site and disaster recovery site

### DISCUSSION THREAD

**Discussion by polaris-alioth** [rabbitmq/rabbitmq-server#3794] [Other] [answered]
Backgroud is  the blog at rabbitmq.com
> https://blog.rabbitmq.com/posts/2020/07/disaster-recovery-and-high-availability-101/

But now a consumer consumes and acknowledges messages m1 and m2 on DC1, and new messages m3 and m4 are published. The quorum queue leader synchronously replicates the new enqueues and the acknowledgements. Federation routes the m3 and m4 messages to DC2 but not the acks (federation does message routing only). Now m1 and m2 only exist on DC2, having been consumed on DC1. If DC1 went down and we failed over to DC2, m1 and m2 would be consumed again.

Is there any setting to auto consume m1 and m2  in DC2 besides message TTL policies and queue length limits ?

**Comment by yaronp68** [comment, marked as answer]
There is a Warm Standby Replication solution that does even more than that. It's a commercial solution as part of VMWare Tanzu RabbitMQ.
This solution does basically 2 things:
- Syncs the definitions between the Active and Standby clusters
- Replicates every message written to a quorum queue into a local replication log on the active
- Any number of standby clusters can register to follow and will consume the replication log into a local replication log on the standby
- The replication log contains also metrics that allows the standby to know which messages were consumed on the active
- When the Standby cluster is promoted to become Active, only the non processed messages will be enqueued to its queues
